lib/cocoapods-downloader/remote_file.rb

```diff
@@ -58,6 +58,28 @@ def should_flatten?
 
       def type_with_url(url)
         case URI.parse(url).path
+        when /\.zip$/
```

Comment:
could this get pulled out into a separate function so the `case` logic isn't duplicated in `type_with_url_query`? maybe `type_with_extension` and then call that here and in `type_with_url_query`